File: doc/inventory-system/backend/internal/model/inventory.go
package model

// Inventory 库存表
type Inventory struct {
	BaseModel
	MaterialCode  string  `gorm:"type:varchar(50);uniqueIndex;not null" json:"material_code"`
	MaterialName  string  `gorm:"type:varchar(200);not null" json:"material_name"`
	Category      string  `gorm:"type:varchar(50);not null" json:"category"`
	Specification string  `gorm:"type:varchar(200);default:''" json:"specification"`
	Unit          string  `gorm:"type:varchar(20);not null" json:"unit"`
	Quantity      float64 `gorm:"type:decimal(12,2);not null;default:0" json:"quantity"`
	SafetyStock   float64 `gorm:"type:decimal(12,2);default:0" json:"safety_stock"`
	Location      string  `gorm:"type:varchar(200);default:''" json:"location"`
	Status        int8    `gorm:"type:tinyint;not null;default:1" json:"status"`
	Remark        string  `gorm:"type:varchar(1000);default:''" json:"remark"`
	Version       int     `gorm:"type:int;not null;default:1" json:"version"`
	IsDeleted     int8    `gorm:"type:tinyint;not null;default:0" json:"is_deleted"`
	CreatedBy     uint    `gorm:"not null" json:"created_by"`
	UpdatedBy     *uint   `gorm:"default:null" json:"updated_by"`

	Attachments []Attachment `gorm:"foreignKey:InventoryID" json:"attachments,omitempty"`
	CreatorName string       `gorm:"-" json:"creator_name,omitempty"`
	Warning     bool         `gorm:"-" json:"warning"`
}

func (Inventory) TableName() string {
	return "inventories"
}

const (
	CategoryRawMaterial     = "raw_material"
	CategoryFinishedProduct = "finished_product"
	CategorySparePart       = "spare_part"
	CategoryOther           = "other"
)

var ValidCategories = []string{
	CategoryRawMaterial,
	CategoryFinishedProduct,
	CategorySparePart,
	CategoryOther,
}

func IsValidCategory(category string) bool {
	for _, c := range ValidCategories {
		if c == category {
			return true
		}
	}
	return false
}

File: doc/inventory-system/frontend/src/types/inventory.ts
// 库存相关类型定义

export interface Inventory {
  id: number;
  material_code: string;
  material_name: string;
  category: string;
  specification: string;
  unit: string;
  quantity: number;
  safety_stock: number;
  location: string;
  status: number;
  remark: string;
  version: number;
  is_deleted: number;
  created_by: number;
  updated_by: number | null;
  created_at: string;
  updated_at: string;
  creator_name?: string;
  warning: boolean;
  attachments?: Attachment[];
}

export interface Attachment {
  id: number;
  inventory_id: number;
  file_name: string;
  display_name: string;
  file_path: string;
  file_size: number;
  file_type: 'image' | 'document';
  mime_type: string;
  extension: string;
  thumbnail_path: string;
  uploaded_by: number;
  uploader_name?: string;
  created_at: string;
  updated_at: string;
}

export interface CreateInventoryRequest {
  material_code: string;
  material_name: string;
  category: string;
  specification?: string;
  unit: string;
  quantity: number;
  safety_stock?: number;
  location?: string;
  status?: number;
  remark?: string;
}

export interface UpdateInventoryRequest {
  material_name?: string;
  category?: string;
  specification?: string;
  unit?: string;
  quantity?: number;
  safety_stock?: number;
  location?: string;
  status?: number;
  remark?: string;
  version: number;
}

export interface InventoryQuery {
  page?: number;
  page_size?: number;
  keyword?: string;
  category?: string;
  status?: number;
  min_quantity?: number;
  max_quantity?: number;
  location?: string;
  show_deleted?: boolean;
}

// 分类选项
export const CATEGORY_OPTIONS = [
  { value: 'raw_material', label: '原料' },
  { value: 'finished_product', label: '成品' },
  { value: 'spare_part', label: '备件' },
  { value: 'other', label: '其他' },
];

// 分类映射
export const CATEGORY_MAP: Record<string, string> = {
  raw_material: '原料',
  finished_product: '成品',
  spare_part: '备件',
  other: '其他',
};
